README.md

```diff
@@ -127,7 +127,304 @@ The Provena repo contains a dedicated integration testing directory, separate fr
+
+```text
+config - Configuration management tool for interacting with a private configuration repository
+
+Usage:
+  config NAMESPACE STAGE [OPTIONS]
+  config --help | -h
+  config --version | -v
+
+Options:
+  --target, -t REPO_CLONE_STRING
+    The repository clone string
+
+  --repo-dir, -d PATH
+    Path to the pre-cloned repository
+
+  --help, -h
+    Show this help
+
+  --version, -v
+    Show version number
+
+Arguments:
+  NAMESPACE
+    The namespace to use (e.g., 'rrap')
+
+  STAGE
+    The stage to use (e.g., 'dev', 'stage')
+
+Environment Variables:
+  DEBUG
+    Set to 'true' for verbose output
+```
+
+The central idea of this configuration approach is that each namespace/stage combination contains a set of files, which are gitignored by default in Provena, which are 'merged' into the user's clone of the Provena repository, allowing temporary access to private information without exposing it in git. 
+
+Sample configuration files are provided in various spots to help get started, including the [template config repo](https://github.com/provena/provena-config-repo-template).
+
+### Config path caching
+
+The script builds in functionality to cache the repo which makes available a given namespace/stage combination. These are stored in `env.json` and has a structure like so:
+
+```json
+{
+  "namespace": {
+    "stage1": "git@github.com:org/repo.git",
+    "stage2": "git@github.com:org/repo.git",
+    "stage3": "git@github.com:org/repo.git"
+  }
+}
+```
+
+This saves using the `--target` option on every `./config` invocation. You can share this file between team members, but we do not recommend committing it to your repository.
+
+## Config definitions
+
+**Namespace**: This is a grouping that we provide to allow you to separate standalone sets of configurations into distinct groups. For example, you may manage multiple organisation's configurations in one repo. You can just use a single namespace if suitable.
+
+**Stage**: A stage is a set of configurations within a namespace. This represents a 'deployment' of Provena.
+
+## Config repository
+
+The config repository is structured carefully, a copy of the README of the [sample config repository](https://github.com/provena/provena-config-repo-template) is included below:
+
+### Purpose
+
+This repository contains configuration files for the Provena project. It serves as a centralized location for managing configuration across different environments and components of the Provena system. This configuration is a mix of sensitive and non-sensitive information, but it is managed centrally to simplify this process.
+
+**Warning: while this repo is private, please don't commit secrets to it, instead commit references/IDs for those secrets stored in AWS Secret Manager**
+
+### `cdk.context.json`
+
+This repo does not contain sample `cdk.context.json` files, but we recommend including this in this repo to make sure deployments are deterministic. This will be generated upon first CDK deploy.
```

Comment:
Are you refering to provena/provena or the users private config repo when using 'This repo' at the sentance beginning?